Frontend/CO.CDP.OrganisationApp/Pages/Forms/DynamicFormsPage.cshtml.cs

```diff
@@ -164,6 +184,26 @@ private bool ValidateTextAnswer()
         return true;
     }
 
+    private bool ValidateDateAnswer()
+    {
+        if (!ModelState.IsValid)
+        {
+            ModelState.AddModelError("Answer", "Please enter a valid date.");
+            return false;
+        }
+
+        if (string.IsNullOrEmpty(FinancialDay) || string.IsNullOrEmpty(FinancialMonth) || string.IsNullOrEmpty(FinancialYear))
+        {
+            ModelState.AddModelError("Answer", "Please enter the date.");
+            return false;
+        }
```

Comment:
This should only be triggered if the question is required (isRequired flag on the question).

Reply:
@jakzal added check